### PR TITLE
Support Java 8 based relations

### PR DESCRIPTION
**Pull request by maxbechtold**
This change enables Usus to detect class (and package) relations based on type references through Lambda expressions or method references (as described in issue #55).

The key changes were using the JLS 8 based AST, which is provided by the Eclipse JDT version 3.10.0. This dependency update requires Luna or newer as the Eclipse platform.

### CODE REVIEW

**Review by marcphilipp — COMMENTED** [review]
As far as I can tell, the changes look good. I added one inline comment.

**Comment by NicoleRauch** [comment]
Looks good to me as well (and I have the same question as @marcphilipp ;-) )

File: org.projectusus.core.test/javaresources/java8relations/A.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.function.Supplier;
+
+public class A {
+
+    public static Object aMethod() {
+        return new Object();
+    }
+
+    public void lambdaReference() {
+        // Supplier<?> supplier = () -> new B();
```
[the remark this answers is not in the document]
Comment (maxbechtold) [reply org.projectusus.core.test/javaresources/java8relations/A.java]: No, it was an oversight. Fixed

**Comment by maxbechtold** [comment]
Thanks for your finding, indeed this slipped my attention. I verified the test behaved as expected and forgot to revert the test file.

**Review by marcphilipp — APPROVED** [review]
👍

**Comment by maxbechtold** [comment]
Merged with consent from @marcphilipp

**Pull request merged**